garageScript/c0d3-app
# feat(discord): Refactor rendering conditions and handle already used account error

## Problem
When a user tries to connect their C0D3 account with a Discord account and a different Discord account is already connected, the `error` parameter with the value `connected` is added to the path `/discord/success`. While this indicates that there is an issue, it does not provide a clear explanation for the user.

To improve the user experience and make it easier for them to understand and communicate this issue to the engineers if needed, it would be helpful to provide a clear error message explaining why the user encountered this issue and what actions they can take to resolve it.

## Steps taken
1. Display a message that explains what went wrong
2. Add a test case to ensure code coverage
3. Refactored the conditions to only use `errorCode` to maintaining the code easier.

## Screenshot
Some screenshots of the before/after changes

### Before
<img width="768" alt="image" src="https://user-images.githubusercontent.com/35906419/210319298-f53e9d07-ae48-4d98-989f-9e8eefeed003.png">

### After
<img width="735" alt="image" src="https://user-images.githubusercontent.com/35906419/210319335-a9698f8a-b743-46b2-b549-2774c2b5457e.png">

## Testing
1. Connect the `admin` account with Discord
2. Log out
4. Login with `newbie` account
5. Try to connect to Discord
6. Notice how it shows the message in the `Screenshot` section
